src/main/java/ch/njol/skript/ScriptLoader.java

```diff
@@ -846,8 +846,11 @@ public static ScriptInfo unloadScripts(Set<Script> scripts) {
 			script.eventRegistry().events(ScriptUnloadEvent.class)
 					.forEach(event -> event.onUnload(parser, script));
 
-			for (Structure structure : script.getStructures())
-				structure.unload();
+			Stream<Structure> structuresStream = script.getStructures().stream();
+			if (isParallel())
+				structuresStream = structuresStream.parallel();
```

Comment:
This doesn't necessarily respect the setting for how many threads should be used, does it? That is, it's not using the threads from the specific async loading thread group. It'd probably be best to push each structure to unload as a new task for one of the threads to pick up. However, that's probably better saved for a future PR (especially given the instability (non-functionality) of parallel loading)

tldr: i think we'd be best off to leave out this change

Reply:
https://github.com/SkriptLang/Skript/pull/8011

Reply:
Since that was merged, can we make use of the executor now?